src/components/overlay-panel/overlay-panel.controller.ts

```diff
@@ -43,21 +43,26 @@ export class OverlayPanelController implements angular.IController {
   private showDialog = () => {
     this.ctrl.shown = true;
     $('body').addClass('overlay-open');
-    this.$document.bind('keydown keypress', this.closeOnEsc);
+    this.$document.bind('keydown', this.closeOnEsc);
   };
 
   private hideDialog = () => {
     this.ctrl.shown = false;
     $('body').removeClass('overlay-open');
-    this.$document.unbind('keydown keypress', this.closeOnEsc);
+    this.$document.unbind('keydown', this.closeOnEsc);
   };
 
   private closeOnEsc = (event: any) => {
     if (event.which === 27) {
-      event.preventDefault();
-      this.$scope.$evalAsync(() => {
-        this.closePanel();
-      });
+      // Only close this overlay if another handler has not called 
+      // preventDefault() on this event. This means only one overlay/modal will
+      // be closed per escape keydown event.
+      if(!event.isDefaultPrevented()) {
```

Comment:
We should double check browser support here. Is this a jQuery event object? If so, it should be OK

https://api.jquery.com/event.isDefaultPrevented/#event-isDefaultPrevented

The standard is `event.defaultPrevented`, though.

https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented

I'm assuming `$document.bind` will pass a jQuery event. `bind()` seems to be deprecated in jQuery 3.0 incidentally.

http://api.jquery.com/bind/

nit: space between `if` and `(`